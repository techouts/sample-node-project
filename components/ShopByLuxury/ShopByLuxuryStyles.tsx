import styled from "@emotion/styled";
import Box from "@mui/material/Box";
import Typography from "@mui/material/Typography";

export const BackgroundBox = styled(Box)((theme) => ({
  opacity: 0.7,
  position: "absolute",
  display: "grid",
  placeItems: "center",
}));
export const OfferText = styled(Box)((theme) => ({
  position: "absolute",
  display: "grid",
  placeItems: "center",
}));
export const StyledText = styled(Typography)((theme) => ({
  fontSize: "18px",
  fontWeight: 500,
  color: "#FFFFFF",
  lineHeight: "24px",
  fontFamily: "Montserrat",
  "@media (max-width: 600px)": {
    fontSize: "12px",
    lineHeight: "15px",
  },
}));

export const MainCard = styled(Box)((theme) => ({
  margin: "0 9px",
  position: "relative",
  cursor: "pointer",
}));
export const StyledImage = styled.img`
  width: 100%;
  height: 100%;
`;
