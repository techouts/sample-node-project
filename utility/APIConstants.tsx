const NEXT_PUBLIC_JUSPAY_APIURL = process.env.NEXT_PUBLIC_JUSPAY_APIURL;
export const merchant_id = "ssbeauty";
export const oauth_token = "authorizationserver/oauth/token";
export const otp_services = "services/v2_1/ssl/sendOTP/OB";
export const otp_service_verifier =
  "services/v2_1/ssl/users/loginOrSignup?mobileNumber=9640558040&otp=782253&cliend_id=mobile_android";
export const SignUp_client_id = "mobile_android";
export const getPaymentMethods = `${NEXT_PUBLIC_JUSPAY_APIURL}/merchants/${merchant_id}/paymentmethods?options.add_emandate_payment_methods=false&options.check_wallet_direct_debit_support=false&options.add_supported_reference_ids=false`;
export const CardBin = `${NEXT_PUBLIC_JUSPAY_APIURL}/cardbins/`;
export const CardTokenization = `${NEXT_PUBLIC_JUSPAY_APIURL}/card/tokenize`;
export const CardTransaction = `${NEXT_PUBLIC_JUSPAY_APIURL}/txns`;
export const QyPaymentStatus = `${NEXT_PUBLIC_JUSPAY_APIURL}/order/payment-status?merchant_id=ssbeauty&order_id=`;
export const paymentStatus =
  "/order/payment-status?merchant_id=ssbeauty&order_id=";
export const GOOGLE_MAP_DIRECTIONS_API = (
  origin: string,
  destination: string
) =>
  `${process.env.NEXT_PUBLIC_GOOGLE_MAP_URL}?api=1&origin=${origin}&destination=${destination}&travelmode=car`;
  export const UNBXD_HOST = process.env.NEXT_UNBXD_HOST;

  export const IS_UNBXD_ENABLED = process.env.NEXT_PUBLIC_IS_UNBXD_ENABLED === "true";

  export const MIDDLEWARE_UNBXD = process.env.NEXT_PUBLIC_MIDDLEWARE_UNBXD_SEARCH;

  export const IS_UNBXD_MIDDLEWARE_ENABLED = process.env.NEXT_PUBLIC_IS_UNBXD_MIDDLEWARE_ENABLED === "true"

  const NEXT_PUBLIC_D365_APP_ID_GUEST = process.env.NEXT_PUBLIC_D365_APP_ID_GUEST;

  const NEXT_PUBLIC_D365_APP_ID_LOGGED_IN = process.env.NEXT_PUBLIC_D365_APP_ID_LOGGED_IN;

  const NEXT_PUBLIC_D365_ORG_ID = process.env.NEXT_PUBLIC_D365_ORG_ID;
  
  const NEXT_PUBLIC_D365_ORG_URL = process.env.NEXT_PUBLIC_D365_ORG_URL;
  export const Google_API_KEY = "AIzaSyCBo5PdGTw5Z0BS6hTboacbxkMC7vIZ6FY";

  export {NEXT_PUBLIC_D365_APP_ID_LOGGED_IN, NEXT_PUBLIC_D365_APP_ID_GUEST, NEXT_PUBLIC_D365_ORG_ID, NEXT_PUBLIC_D365_ORG_URL}
  export const IS_RECAPTCHA_ENABLED = process.env.NEXT_PUBLIC_IS_RECAPTCHA_ENABLED === "true";