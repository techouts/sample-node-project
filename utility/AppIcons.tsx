export const Camera_ICON = "Camera_icon";
export const EDIT_IMAGE_ICON = "EDIT_IMAGE";
export const InfoCircle_Grey_ICON = "InfoCircle_Grey";
export const Profile_Add_ICON = "Profile_Add";

export const TIER0_ICON = "TIER0_LOGO";
export const TIER1_ICON = "TIER1_LOGO";
export const TIER2_ICON = "TIER2_LOGO";
export const TIER3_ICON = "TIER3_LOGO";
export const TIER4_ICON = "TIER4_LOGO";

export const GET_APP_LOGO_HEADER = "Get_App";
export const GIFT_CARD_ICON = "Gift_Card";
export const STORES_AND_EVENTS_ICON = "Stores_And_Events";
export const HELP_ICON = "Help";
export const USER_SIGNIN_ICON = "user_signin";
export const LOCATION_ICON = "Location";
export const DOWN_ARROW_ICON = "ARROWDOWN";
export const DELETE_ICON = "Delete_icon";
export const ADD_ADDRESS_ICON = "Add_address";
export const CHECKED_ICONS = "CHECKED_ICON";
export const UNCHECKED_ICONS = "UNCHECKED_ICON";
export const BeautyProfilecheck_ICON = "BeautyProfilecheck";
export const EDIT_PINK_ICON = "EditIcon_PinkColour";
export const CHECKED_RADIO_ICON = "Selected_Radio_Button";
export const UNCHECKED_RADIO_ICON = "Unselected_Radio_Button";
export const VERIFIED_IMAGE_ICON = "VERIFIED_IMAGE";
export const TICK_CIRCLE_WARNING = "TickCircleWarning";
export const COD_ICON = "CODImage";
export const CART_TRUCK_ICON = "CartTruckImage";
export const EMPTY_WISHLIST_COLOUR = "Wishlist_Empty_Colour";
export const NORMAL_SEARCH_ICON = "SEARCH_NORMAL";
export const FILTER_SETTINGS_ICON = "Filter_settings";
export const REFRESH_ICONS = "Refresh_icon";
export const SPEAKER_ICONS = "Speaker_icon";
export const CARD_PAYMENT_IMAGE = "CardPaymentImage";
export const AIRTEL_MOBILE_IMAGE = "Airtel_image_mobile";
export const MOBIWICK_MOBILE_IMAGE = "Mobiwick_mobile_image";
export const FREECHARGE_MOBILE_IMAGE = "Freecharge_mobile_image";
export const JIO_MOBILE_IMAGE = "Jio_mobile_image";
export const ZOOM_ICON = "ZoomIconUrl";
export const SHARE_ICONS = "Share_Icon";
export const WISHLIST_ADDED_ICON = "Wishlist_filled_heart";
export const PLAY_VIDEO_ICON = "VideoPlayButton";
export const LEFT_ARROW_ICON = "Left_direction_icon";
export const USER_ACCOUNT_ICON = "account_user_icon"
export const CLOSE_CIRCLE_ICON = "close-circle";
export const LIKE_ICONS = "LIKE_ICON";
export const UNLIKE_ICONS = "UNLIKE_ICON";
export const FILLED_TICK_CIRLE_ICON = "Filled_tick_circle";
export const Closecircle_BlackColour_ICON = "Closecircle_BlackColour";
export const current_location_pointer_ICON = "current_location_pointer";
export const Empty_wishlist_webicon_icon = "Empty_wishlist_webicon";
export const Empty_wishlist_appbgicon_icon = "Empty_wishlist_appbgicon";
export const CLOSE_CIRCLE_IMAGE = "close_circle";
export const DISCOUNTS_ICON = "discount_shape";
export const SORT_ICON = "SORT_ARROW";
export const EMPTY_STAR_ICON = "EmptyStar";
export const TROPHY_ICONS = "Trophy_icon";
export const GPS_LOCATION_ICON = "Gps_Pink_Icon";
export const HEADER_SEARCH_ICON = "headerSearchIcon";
export const HEADER_WISHLIST_ICON = "headerWishlistIcon";
export const HEADER_CART_ICON = "headerCartIcon";
export const BRAND_UP_ARROW = "BRAND_UP_ARROW";
export const BRAND_DOWN_ARROW = "BRAND_DOWN_ARROW";
export const HEADER_BACK_ARROW = "headerNavigationIcon";
export const INFO_ICON = "INFO_ICON";
export const FALL_BACK_IMAGE_PRODUCT = "PRODUCT_FALLBACK_URL";
export const Error_Image = "ErrorImage";
export const No_RESULTS_SEARCH_ICON = "noResultsSearchIcon";
export const RECENT_SEARCH_ICON =
  "https://storage.googleapis.com/images_cms_sit_ssb/recent_clock_bde48d7c4e_1_986324a98d.svg";