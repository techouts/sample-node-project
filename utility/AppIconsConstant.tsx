import { useRecoilState } from "recoil";
import { SSBLogos } from "../recoilstore";

export const AppIcons = (image: string) => {
  const [SSBeautyLogos] = useRecoilState(SSBLogos);

  const imageIcon = SSBeautyLogos?.appLogos?.find(
    (item: any) => item?.name == image
  );
  return imageIcon;
};
