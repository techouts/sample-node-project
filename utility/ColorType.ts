import { type } from "os";

type RGB = `rgb(${number}, ${number}, ${number})`;
type RGBA = `rgba(${number}, ${number}, ${number}, ${number})`;
type HEX = `#${string}`;

export type ColorType = RGB | RGBA | HEX;