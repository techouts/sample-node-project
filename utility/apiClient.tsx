import axios, { AxiosRequestConfig, AxiosResponse } from "axios";
import { callErrorEvent } from "./ErrorEventAnalytics";
import { fetchMiddleWareAccessToken } from "./MiddelwareAccessWrapper";

const GRAPHQL_URL = process.env.NEXT_PUBLIC_GRAPHQL_URL!;
const MIDDLEWARE_URL = process.env.NEXT_PUBLIC_SSO_API_GATEWAY_URL!;
const isMiddlewareSecurityEnabled =
  process.env.NEXT_PUBLIC_IS_MIDDLEWARE_SECURITY_ENABLED === "true";

const axiosInstance = axios.create();

async (config: AxiosRequestConfig<any>) => {
  if (!config.url?.includes("/middleware")) return config;

  const token = localStorage.getItem("accessToken");
  const updatedHeaders: AxiosRequestConfig["headers"] = {
    ...config.headers,
    authorization: token ? `Bearer ${token}` : "",
  };
  if (isMiddlewareSecurityEnabled) {
    let mToken = localStorage.getItem("m-token");
    if (mToken && mToken !== "") {
      updatedHeaders["x-auth-token"] = `Bearer ${mToken}`;
    } else {
      const mTokenRes = await fetchMiddleWareAccessToken();
      if (mTokenRes?.data?.access_token) {
        updatedHeaders[
          "x-auth-token"
        ] = `Bearer ${mTokenRes?.data?.access_token}`;
      }
    }
  }

  return { ...config, headers: updatedHeaders };
};

axiosInstance.interceptors.response.use(
  (response: AxiosResponse) => {
    return response;
  },
  async (error: any) => {
    if (error.response) {
      const { message, status } = error.response.data || {};
      console.error(`[Error]: ${message}, Status: ${status}`);

      let params = {
        error_message: message || error.message,
        status: status || error.response?.status,
        item_name: GRAPHQL_URL,
      };

      callErrorEvent(params);
    } else if (error.request) {
      console.error("[Network error]: No response received");
    } else {
      console.error(`[Error]: ${error.message}`);
    }
    return Promise.reject(error);
  }
);

// Retry logic for requests
const retryRequest = async (
  config: AxiosRequestConfig<any>,
  retries = 3
): Promise<AxiosResponse | void> => {
  let lastError;

  for (let i = 0; i < retries; i++) {
    try {
      return await axiosInstance(config);
    } catch (error: any) {
      lastError = error;

      if (error.response?.status === 401 || error.response?.status === 403) {
        console.warn(
          `Attempt ${i + 1}: Received ${
            error.response.status
          }, trying to refresh token...`
        );
        const mTokenRes = await fetchMiddleWareAccessToken();
        const mToken = mTokenRes?.data?.access_token;

        if (!mToken) {
          console.error("Token refresh failed. Aborting retries.");
          break;
        }
        localStorage.setItem("m-token", mToken);
        config = {
          ...config,
          headers: {
            ...(config.headers || {}),
            "x-auth-token": `Bearer ${mToken}`,
          },
        };

        continue;
      }

      console.error(
        `Request failed with status ${error.response?.status}. Retrying...`
      );
    }
  }

  return Promise.reject(lastError);
};

// Function to send requests
const sendRequest = async (
  query: string,
  isCMSQuery = false,
  middleWareURL = false,
  urlKey?: string
): Promise<AxiosResponse | void> => {
  const url = urlKey
    ? `${urlKey}?query=${query}`
    : isCMSQuery
    ? query
    : middleWareURL
    ? `${MIDDLEWARE_URL}?query=${query}`
    : `${GRAPHQL_URL}?query=${query}`;

  const config: AxiosRequestConfig = {
    method: "get",
    url: url,
    headers: {},
  };

  try {
    return await retryRequest(config);
  } catch (error: any) {
    console.error(`Request failed: ${error}`);
    return Promise.reject(error);
  }
};

export default sendRequest;
