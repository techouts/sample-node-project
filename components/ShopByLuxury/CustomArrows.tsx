import { NEXTARROW, PREVARROW } from "../../utility/Constants";

export function CustomPrevArrow(props: any) {
  const { className, style, onClick, cssData, isPrev } = props;
  return (
    <div
      className={className}
      style={{
        ...style,
        ...cssData,
        display: "block",
        background:
          `url(${PREVARROW}) no-repeat`,
        filter: `grayscale(${isPrev ? "0%" : "100%"})`,
        opacity: isPrev ? "1" : "0.2",
        cursor: isPrev ? "pointer" : "auto",
        backgroundSize: 'contain'
      }}
      onClick={onClick}
    />
  );
}
export function CustomNextArrow(props: any) {
  const { className, style, onClick, cssData, isNext } = props;
  return (
    <div
      className={className}
      style={{
        ...style,
        ...cssData,
        display: "block",
        background:
          `url(${NEXTARROW}) no-repeat`,
        filter: `grayscale(${isNext ? "0%" : "100%"})`,
        opacity: isNext ? "1" : "0.2",
        cursor: isNext ? "pointer" : "auto",
        backgroundSize: 'contain'
      }}
      onClick={onClick}
    />
  );
}
