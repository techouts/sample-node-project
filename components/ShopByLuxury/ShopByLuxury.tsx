import ExperienceLuxuryStoreSchema from "../../schemas/ShopByLuxurySchema";
import { CustomPrevArrow, CustomNextArrow } from "./CustomArrows";
import {
  StyledText,
  OfferText,
  MainCard,
  StyledImage,
  BackgroundBox,
} from "./ShopByLuxuryStyles";
import Carousel from "../../HOC/Carousel/Carousel2";
import Box from "@mui/material/Box";
import Grid from "@mui/material/Grid";
import { useRef, useState, useEffect } from "react";
import Title from "../../HOC/Title/Title";
import { useMobileCheck } from "../../utility/isMobile";
import { ReplaceImage } from "../../utility/ReplaceImage";
import { widget_type, event_type } from "../../utility/GAConstants";
import ViewEvent from "../../utility/viewEvent";
import triggerGAEvent from "../../utility/GaEvents";
import {
  ShopbyluxuryViewArray,
  viewArray,
} from "../../utility/ViewEvenItemArray";
import { useRouter } from "next/router";
import { gaActionEvent } from "../../utility/gaActionEvent";
import { useInView } from "react-intersection-observer";
import triggerGaViewPromotion from "../../utility/GAViewPromotion";
import { Cookies } from "react-cookie";
import useStorage from "../../utility/useStoarge";
import TriggerGaViewPromotion from "../../utility/GAViewPromotion";

const ShopByLuxury = ({
  bgColor,
  bgPadding,
  items,
  title,
  position,
  id,
  titleColor,
  __component,
  promotion_id,
  promotion_name,
  viewport,
}: ExperienceLuxuryStoreSchema) => {
  const cookie = new Cookies();
  const { getItem } = useStorage();
  const carouselRef = useRef<HTMLDivElement | null>(null);

  useEffect(() => {
    const carouselElement = carouselRef.current;
    let startX = 0;
    let startY = 0;
    let isSwiping = false;

    const handleTouchStart = (e: TouchEvent) => {
      startX = e.touches[0].clientX;
      startY = e.touches[0].clientY;
      isSwiping = false;
    };

    const handleTouchMove = (e: TouchEvent) => {
      const dx = e.touches[0].clientX - startX;
      const dy = e.touches[0].clientY - startY;

      if (Math.abs(dx) > Math.abs(dy)) {
        isSwiping = true;
        // Prevent vertical scroll during horizontal swipe
        e.preventDefault();
      }
    };

    const handleTouchEnd = () => {
      isSwiping = false;
    };

    if (carouselElement) {
      carouselElement.addEventListener("touchstart", handleTouchStart);
      carouselElement.addEventListener("touchmove", handleTouchMove, {
        passive: false,
      });
      carouselElement.addEventListener("touchend", handleTouchEnd);
    }

    return () => {
      if (carouselElement) {
        carouselElement.removeEventListener("touchstart", handleTouchStart);
        carouselElement.removeEventListener("touchmove", handleTouchMove);
        carouselElement.removeEventListener("touchend", handleTouchEnd);
      }
    };
  }, []);

  const isMobile = useMobileCheck();

  const viewEventWrapper = useRef();
  const [prevArrow, setPrevArrow] = useState(false);
  const [nextArrow, setNextArrow] = useState(
    items?.length === 3 ? false : true
  );
  const [currentIndex, setCurrentIndex] = useState(0);
  const initialSlides = isMobile ? 2.5 : 3;
  const [completedIndex, setCompletedIndex] = useState(initialSlides);
  const { ref, inView } = useInView({ triggerOnce: true });
 
  const handleArrowData = (index: number, eventName: string) => {
    triggerGAEvent(
      {
        item_name: "na",
        item_id: `${__component}_${position}_${index + 1}`,
        widget_type: widget_type,
        component_id: id,
        item_type: "na",
        widget_title: __component,
        widget_description: "na",
        widget_postion: position,
        no_of_items: items?.length,
        index: index + 1,
        item_brand: "na",
        item_category: "na",
      },
      eventName
    );
  };
  const userSettings = {
    arrows: true,
    draggable: true,
    dots: false,
    infinite: false,
    speed: 500,
    initialSlide: 0,
    slidesToShow: 3,
    slidesToScroll: 1,
    swipeToSlide: true,
    prevArrow: (
      <CustomPrevArrow
        isPrev={prevArrow}
        cssData={{ width: "36px", height: "36px", left: "-40px" }}
      />
    ),
    nextArrow: (
      <CustomNextArrow
        isNext={nextArrow}
        cssData={{ width: "36px", height: "36px", right: "-40px" }}
      />
    ),
    responsive: [
      {
        breakpoint: 1200,
        settings: {
          slidesToShow: 3,
          slidesToScroll: 1,
        },
      },
      {
        breakpoint: 900,
        settings: {
          slidesToShow: 2.2,
          slidesToScroll: 1,
          arrows: false,
        },
      },
      {
        breakpoint: 600,
        settings: {
          slidesToShow: 2.2,
          slidesToScroll: 1,
          arrows: false,
        },
      },
    ],
  };

  let newVisibleProducts: any = [];

  const trackGAEvent = (index: number) => {
    const roundedIndex = Math.round(index);
    const startIndex = roundedIndex;
    const endIndex = Math.min(roundedIndex + initialSlides, items.length);

    const visibleProducts = items.slice(startIndex, endIndex);

    const newItems = visibleProducts.filter(
      (item) => !newVisibleProducts.includes(item)
    );

    newVisibleProducts = [...newVisibleProducts, ...newItems];

    const datasLayer = {
      previous_screen_name: "na",
      visitor_type: cookie.get("accessToken") ? "loggedin" : "Guest",
      user_id: getItem("customer_ref", "local") || "na",
      creative_name: `${newItems
        ?.map((item: any) => `${item?.creative_name}-outer SBL`)
        .join("|")}`,
      creative_slot: `${newItems
        ?.map((item: any) => `${item?.creative_slot}-outer SBL`)
        .join("|")}`,
      items: newItems?.map((item: any) => ({
        promotion_id: `${promotion_id}-inner SBL`,
        promotion_name: `${promotion_name}-inner SBL`,
        creative_name: `${item?.creative_name}-inner SBL`,
        creative_slot: `${item?.creative_slot}-inner SBL`,
      })),
    };
    TriggerGaViewPromotion("view_promotion", datasLayer);
  };

  return (
    <>
      <Box
        ref={viewEventWrapper}
        bgcolor={bgColor}
        p={isMobile ? "0 0 25px 16px" : bgPadding}
      >
        <Box component="div" pb={isMobile ? "10px" : "18px"}>
          <Title
            interface={{
              title: title,
              bgColor: bgColor,
              titleColor: titleColor,
            }}
          />
        </Box>
        <Grid ref={ref} style={{ touchAction: "pan-y" }}>
          {inView && (
            <Carousel
              Component={LuxuryCard}
              items={items}
              settings={userSettings}
              callBack={undefined}
              isVertical={undefined}
              componentvalue={__component}
              widgetposition={position}
              promotion_id={promotion_id}
              promotion_name={promotion_name}
              trackGAEvent={trackGAEvent}
            />
          )}
        </Grid>
      </Box>
    </>
  );
};
const LuxuryCard = (props: any) => {
  const {
    path,
    isNewTab,
    textBgColor,
    text,
    imageUrl,
    id,
    componentvalue,
    widgetposition,
    dataSet,
    viewport,
  } = props;

  const isMobile = useMobileCheck();
  const router = useRouter();
  const cookie = new Cookies();
  const { getItem } = useStorage();

  const handleClick = () => {

    const datasLayer = {
      previous_screen_name: "na",
      visitor_type: cookie.get("accessToken") ? "loggedin" : "Guest",
      user_id: getItem("customer_ref", "local") || "na",
      creative_name: `${props?.creative_name}-outer SBL`,
      creative_slot: `${props?.creative_slot}-outer SBL`,
      items: [
        {
          promotion_id: `${props?.promotion_id}-inner SBL`,
          promotion_name: `${props?.promotion_name}-inner SBL`,
          creative_name: `${props?.creative_name}-inner SBL`,
          creative_slot: `${props?.creative_slot}-inner SBL`,
        },
      ],
    };

    triggerGaViewPromotion("select_promotion", datasLayer);
    if (isNewTab) {
      path && window?.open(path);
    } else {
      path && router?.push(path);
    }
  };
  return (
    <>
      <MainCard onClick={() => handleClick()} style={{ touchAction: "pan-y" }}>
        <StyledImage
          src={`${ReplaceImage(imageUrl)}`}
          alt="LuxuryCardImage"
        ></StyledImage>
        <BackgroundBox
          height={isMobile ? "44px" : "72px"}
          color={"white"}
          bgcolor={textBgColor}
          textAlign={"center"}
          width={isMobile ? "calc(100% - 6px)" : "calc(100% - 16.5px)"}
          bottom={isMobile ? "3px" : "7.5px"}
          mx={isMobile ? "3px" : "8px"}
        >
          <StyledText sx={{ visibility: "hidden" }}>{text}</StyledText>
        </BackgroundBox>
        <OfferText
          height={isMobile ? "44px" : "72px"}
          color={"white"}
          bgcolor="transparent"
          textAlign={"center"}
          width={isMobile ? "calc(100% - 6px)" : "calc(100% - 16.5px)"}
          bottom={isMobile ? "3px" : "7.5px"}
          mx={isMobile ? "3px" : "8px"}
        >
          <StyledText>{text}</StyledText>
        </OfferText>
      </MainCard>
    </>
  );
};
export default ShopByLuxury;
