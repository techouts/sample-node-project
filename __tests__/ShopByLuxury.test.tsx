import { render } from "@testing-library/react";
import { ColorType } from "../utility/ColorType";
import ShopByLuxury from "../components/ShopByLuxury/ShopByLuxury";
const color: ColorType = "#0000";

const shopByLuxuryData = {
  id: 3,
  __component: "widget.shop-by-luxury",
  bgColor: color,
  bgPadding: "3% 5%",
  title: "EXPERIENCE LUXURY",
  titleColor: "#FFFFFF",
  display: null,
  items: [
    {
      id: 9,
      imageUrl:
        "https://ssbeauty-cms.s3.ap-south-1.amazonaws.com/Experience_Lux1_4df3d329a8.png",
      text: "Bestseller Starting At Rs 1,540 ",
      path: "/brand/faces-canada",
      isNewTab: false,
      textBgColor: color,
      border: "",
      title: "string",
      position: 12,
      __component: "string",
    },
  ],
  position: 12,
};

describe("Shop By Luxury ", () => {
  it("renders a shop by luxury component", () => {
    const { container } = render(<ShopByLuxury {...shopByLuxuryData} />);
    expect(container).toMatchSnapshot();
  });
});
