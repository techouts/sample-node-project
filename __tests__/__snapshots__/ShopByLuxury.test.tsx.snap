// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop By Luxury  renders a shop by luxury component 1`] = `
<div>
  <div
    class="MuiBox-root css-18hz1cm"
  >
    <div
      class="MuiBox-root css-l57rop"
    >
      <div
        class="MuiBox-root css-83no3v"
      >
        EXPERIENCE LUXURY
      </div>
    </div>
    <div
      class="MuiGrid-root css-vj1n65-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-8atqhb"
      >
        <div
          class="MuiBox-root css-t7vv4g"
        >
          <div
            class="slick-slider slick-initialized"
          >
            
            <div
              class="slick-list"
            >
              <div
                class="slick-track"
                style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
              >
                <div
                  aria-hidden="false"
                  class="slick-slide slick-active slick-current"
                  data-index="0"
                  style="outline: none; width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="MuiBox-root css-1d8o3k"
                    >
                      <img
                        alt="LuxuryCardImage"
                        class="css-s08p0c"
                        src="https://images-cms.sit.shopper-stop.in/Experience_Lux1_4df3d329a8.png"
                      />
                      <div
                        class="MuiBox-root css-12pq0yk"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-1uk6g1r-MuiTypography-root"
                        >
                          Bestseller Starting At Rs 1,540 
                        </p>
                      </div>
                      <div
                        class="MuiBox-root css-e67j02"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-16dmanc-MuiTypography-root"
                        >
                          Bestseller Starting At Rs 1,540 
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            
            
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
